src/main/java/frc/robot/commands/auto/TurnAngle.java

```diff
@@ -0,0 +1,83 @@
+/* ==================================================
+ * Authors: Lucas Jacobs
+ *
+ * --------------------------------------------------
+ * Description:
+ * Makes the robot turn a given angle in degrees using proportional speed
+ * ================================================== */
+
+package frc.robot.commands.auto;
+
+import frc.robot.Robot;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class TurnAngle extends CommandBase
+{
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+
+  // VARIABLES //
+
+  private Robot robot;
+  private double speed;
+  private double degrees;
+  private double targetAngle;
+
+  // CONSTANTS //
+  private static final double MIN_TURN_SPEED = .3;
+  private static final double ANGLE_THRESHOLD = 10;
+
+  // CONSTRUCTOR //
+
+  public TurnAngle(Robot robot, double speed, double degrees)
+  {
+    this.robot = robot;
+    this.speed = speed;
+    this.degrees = degrees;
+
+    // subsystems that this command requires
+    addRequirements(robot.drivetrain);
+  }
+
+  // METHODS //
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize()
+  {
+    targetAngle = SmartDashboard.getNumber("Gyro Angle", 0) + degrees;
+
+    SmartDashboard.putString("Robot Mode:", "Turn Angle");
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute()
+  {
+    double currentAngle = Math.abs(SmartDashboard.getNumber("Gyro Angle", 0));
```

Comment:
If you move the accel and gyro to drivetrain you don't have to read back from the dash every time.  The dash is very slow, you generally only want to use it to post information to the drivers but never to store information for processing.